node/cmd/spy/spy.go

```diff
@@ -185,10 +380,75 @@ func (s *spyServer) SubscribeSignedVAA(req *spyv1.SubscribeSignedVAARequest, res
 	}
 }
 
+// SubscribeSignedVAAByType fields requests for subscriptions. Each new subscription adds a channel and request params (filters)
+// to the map of active subscriptions.
+func (s *spyServer) SubscribeSignedVAAByType(req *spyv1.SubscribeSignedVAAByTypeRequest, resp spyv1.SpyRPCService_SubscribeSignedVAAByTypeServer) error {
+	var fi []spyv1.FilterEntry
+	if req.Filters != nil {
+		for _, f := range req.Filters {
+			switch t := f.Filter.(type) {
+
+			case *spyv1.FilterEntry_EmitterFilter:
+				// validate the emitter address is valid by decoding it
+				_, err := decodeEmitterAddr(t.EmitterFilter.EmitterAddress)
```

Comment:
Can we store the decoded address here rather than dropping it?  Then we don't need to decode it every time in `PublishSignedVAAByType`.

Reply:
I agree, having the `spyv1.FlterEntry` type differ from the VAA's type for the same data is a mess. Previously, spy implemented a third set of structs that held transformed values of the `spyv1.EmitterFilter` values to be similar to the VAA's types ([`filter`](https://github.com/wormhole-foundation/wormhole/blob/b129fcb52dc531bdc7b8459cca9c9ec5d7e2863a/node/cmd/spy/spy.go#L81), [`subscription`](https://github.com/wormhole-foundation/wormhole/blob/b129fcb52dc531bdc7b8459cca9c9ec5d7e2863a/node/cmd/spy/spy.go#L86)) .

I don't think Spy should be in the business of transforming all the `spyv1.FilterEntry` values, handling the bad data, and then holding them in state for when a VAA comes along. So, I updated the proto types (all unused properties so OK to change), to align more closely with the `gossip` messages received by Spy. Now the filter structs of subscribers are held in state, and can be directly compared to the incoming `gossip` VAAs. This gets rid of the duplicate decoding, and a ton of boilerplate.

I also updated the response type of the new Spy endpoint so Spy can just forward the gossip messages along to subscribers without having to create custom response structs. I think the tight coupling is worth it here.